kong/plugins/proxy-cache/handler.lua

```diff
@@ -400,6 +400,7 @@ function ProxyCacheHandler:header_filter(conf)
 
   -- if this is a cacheable request, gather the headers and mark it so
   if cacheable_response(conf, cc) then
+    -- TODO: should this use the kong.conf configured limit?
```

Comment:
I agree.
We have had a similar problem (that you fixed) with the read bodies issue in the sessions plugin and this smells like the same issue.
Should we consider updating the code in globalpatches.lua to enforce the hard limit via the patched function?